docs/design/kuadrant-api-v2.md

```diff
@@ -0,0 +1,178 @@
+# Kuadrant Proposal - API
+
+Kuadrant aims to expose a simple API, yet flexible enough to accomodate API Management requirements.
+
+## Problem Statement
+
+Currently, the [latest](https://github.com/Kuadrant/kuadrant-controller/releases/tag/v0.2.0) release of
+kuadrant API is based on [apiproducts.v1beta1.networking.kuadrant.io](https://github.com/Kuadrant/kuadrant-controller/blob/v0.2.0/doc/architecture.md#api-product-crd)
+and [apis.v1beta1.networking.kuadrant.io](https://github.com/Kuadrant/kuadrant-controller/blob/v0.2.0/doc/architecture.md#api-crd)
+custom kubernetes resource APIs. Together with these CRDs, there is a bunch of kuadrant service
+annotations and labels available in order to provide the kuadrant
+[service discovery](https://github.com/Kuadrant/kuadrant-controller/blob/v0.2.0/doc/service-discovery.md)
+feature. All together, kuadrant provides routing, basic and global rate limiting and, finally,
+global authN (not authZ) features to protect customer owned APIs. While, good enough for a wide set
+of use cases, there are several limitations:
+
+* Rate limit is enforced to all services and endpoints defined in a APIProduct. It cannot be specified route/path wise.
+* Only high level rate limit directives are defined (`global`, `perRemoteIP` and `authenticated`). There is no option to specify with a fine-grained rate limit API.
+* Authentication is enforced to all services and endpoints defined in a APIProduct. It cannot be specified route/path wise.
+* Authentication is limited to `API key` and `OpenID Connect`.
+* Authorization is not available.
+
+## High Level Goals
+
+* Expose simple API, API management oriented, to protect customer owned API's.
+* Simple API can be extended with a fine-grained API to leverage full featured Authorino API and Limitador API.
+* Path/Operation based protection (i.e. authentication and rate limit)
+
+## Proposed Solution
```

Comment:
I wonder would it help to capture a set of use cases that the API will help solve?

``` As an API Provider, I want to be able to apply a set of authenticated rate limiting plans based use specific attributes (for example a JWT claim ) so that I can allow different tiers of access to my API```

``` As an API Provider I want to be able to apply both a set of authenticated and unauthenticated rate limit plans so that I can have a "try for free" tier that doesn't require the user to sign up and so that I can have unauthenticated APIs freely available but protected ```

Reply:
good contribution :+1: